src/distribute_kv_cache/rpc/packet.rs

```diff
@@ -0,0 +1,542 @@
+use std::{collections::HashMap, fmt::Debug, sync::Arc};
+
+use async_trait::async_trait;
+use bytes::{BufMut, BytesMut};
+use tokio::{sync::Mutex, time::Instant};
+use tracing::debug;
+
+use super::{
+    error::RpcError,
+    utils::{get_u64_from_buf, get_u8_from_buf, u64_to_usize},
+};
+
+/// The size of the request header.
+pub const REQ_HEADER_SIZE: u64 = 17;
+/// The size of the response header.
+pub const RESP_HEADER_SIZE: u64 = 17;
+
+/// The Encode trait is used to encode a message structure into a byte buffer.
+pub trait Encode: Sync {
+    /// Encode the message into a byte buffer, this operation will append to buffer
+    /// If you need to encode from start, you should clear the buffer first
+    fn encode(&self, buf: &mut BytesMut);
+
+    /// Get large data with Bytes, disable memory copy
+    fn encode_large_data(&self, _buf: &mut BytesMut) -> (u64, Vec<bytes::Bytes>) {
+        // Default is empty
+        (0, vec![])
+    }
+}
+
+/// The Decode trait is used to message a byte buffer into a data structure.
+#[allow(clippy::unimplemented)]
+pub trait Decode {
+    /// Decode the byte buffer into a data structure
+    fn decode(_buf: &mut BytesMut) -> Result<Self, RpcError>
+    where
+        Self: Sized,
+    {
+        unimplemented!("Not implemented for this type")
+    }
+
+    /// Decode the large data into a data structure
+    fn decode_large_data(_buf: bytes::Bytes) -> Result<Self, RpcError>
+    where
+        Self: Sized,
+    {
+        unimplemented!("Not implemented for this type")
+    }
```

Comment:
Why not separate these two functions into two difference traits as you don't always implement them at the same time?